packages/common/src/types/filterGrammarConversion.ts

```
@@ -0,0 +1,130 @@
+        );
+    }
+    // Operators without values are handled here
+    switch (operator) {
+        case FilterOperator.NULL:
+            return [`null`];
+        case FilterOperator.NOT_NULL:
+            return [`!null`];
+        default:
+            break; // continue
+    }
+    const validValues = values as (string | number)[];
+    const firstValue = validValues[0];
+
+    switch (operator) {
+        case FilterOperator.EQUALS:
+            return validValues?.map((value) => `${value}`);
+        case FilterOperator.NOT_EQUALS:
+            return [`!${firstValue}`];
+        case FilterOperator.INCLUDE:
+            return [`%${firstValue}%`];
+        case FilterOperator.NOT_INCLUDE:
+            return [`!%${firstValue}%`];
+        case FilterOperator.STARTS_WITH:
+            return [`${firstValue}%`];
+        case FilterOperator.ENDS_WITH:
+            return [`%${firstValue}`];
+        case FilterOperator.GREATER_THAN:
+            return [`> ${firstValue}`];
+        case FilterOperator.GREATER_THAN_OR_EQUAL:
+            return [`>= ${firstValue}`];
+        case FilterOperator.LESS_THAN:
+            return [`< ${firstValue}`];
+        case FilterOperator.LESS_THAN_OR_EQUAL:
+            return [`<= ${firstValue}`];
+        case FilterOperator.IN_THE_NEXT:
+        case FilterOperator.IN_THE_PAST:
+            if (isDateFilterRule(filter)) {
+                const settings = filter.settings as DateFilterSettings;
+                const unitOfTime = settings.unitOfTime || UnitOfTime.days;
+
+                let totalValue = Number(firstValue);
+                if (
+                    settings.completed &&
+                    operator === FilterOperator.IN_THE_NEXT
+                ) {
+                    totalValue += 1;
+                }
+                if (
+                    settings.completed &&
+                    operator === FilterOperator.IN_THE_PAST
+                ) {
+                    totalValue -= 1;
```

I would throw `NotImplementedError` when complete=true rather than returning the wrong filter logic.
Filter in UI
`date is in the last 1 completed year` (2024.01.01 - 2024.12.31)
converted to YML (1-1=0)
`date is in the last 0 year` 🤔